aten/src/ATen/cpu/vec256/vec256_float_neon.h

```diff
@@ -0,0 +1,653 @@
+  }
+};
+
+template <> class Vec256<float> {
+private:
+  float32x4x2_t values;
+public:
+  using value_type = float;
+  static constexpr int size() {
+    return 8;
+  }
+  Vec256() {}
+  Vec256(float32x4x2_t v) : values(v) {}
+  Vec256(float val) {
+    values.val[0] = vdupq_n_f32(val);
+    values.val[1] = vdupq_n_f32(val);
+  }
+  Vec256(float val0, float val1, float val2, float val3,
+         float val4, float val5, float val6, float val7) {
+    values.val[0] = {val0, val1, val2, val3};
+    values.val[1] = {val4, val5, val6, val7};
+  }
+  Vec256(float32x4_t val0, float32x4_t val1) {
+    values.val[0] = val0;
+    values.val[1] = val1;
+  }
+  operator float32x4x2_t() const {
+    return values;
+  }
+  template <int64_t mask>
+  static Vec256<float> blend(const Vec256<float>& a, const Vec256<float>& b) {
+    Vec256<float> vec;
+    // 0.
+    vec.values.val[0] =
+      BlendRegs<0, (mask & 0x01)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    vec.values.val[0] =
+      BlendRegs<1, (mask & 0x02)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    vec.values.val[0] =
+      BlendRegs<2, (mask & 0x04)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    vec.values.val[0] =
+      BlendRegs<3, (mask & 0x08)!=0>::impl(
+          a.values.val[0], b.values.val[0], vec.values.val[0]);
+    // 1.
+    vec.values.val[1] =
+      BlendRegs<0, (mask & 0x10)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    vec.values.val[1] =
+      BlendRegs<1, (mask & 0x20)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    vec.values.val[1] =
+      BlendRegs<2, (mask & 0x40)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    vec.values.val[1] =
+      BlendRegs<3, (mask & 0x80)!=0>::impl(
+          a.values.val[1], b.values.val[1], vec.values.val[1]);
+    return vec;
+  }
+  static Vec256<float> blendv(const Vec256<float>& a, const Vec256<float>& b,
+                              const Vec256<float>& mask) {
+    // TODO
+    // NB: This requires that each value, i.e., each uint value,
+    // of the mask either all be zeros or all be 1s.
+    // We perhaps need some kind of an assert?
+    // But that will affect performance.
+    Vec256<float> vec(mask.values);
+    vec.values.val[0] = vbslq_f32(
+        vreinterpretq_u32_f32(vec.values.val[0]),
+        b.values.val[0],
+        a.values.val[0]);
+    vec.values.val[1] = vbslq_f32(
+        vreinterpretq_u32_f32(vec.values.val[1]),
+        b.values.val[1],
+        a.values.val[1]);
+    return vec;
+  }
+  template<typename step_t>
+  static Vec256<float> arange(float base = 0.f, step_t step = static_cast<step_t>(1)) {
+    return Vec256<float>(
+      base,            base +     step, base + 2 * step, base + 3 * step,
+      base + 4 * step, base + 5 * step, base + 6 * step, base + 7 * step);
```

Comment:
This could be vectorized itself.  Currently all these operations are done on scalars.  Imagine _base_, _i_ = [0, 1, 2, 3, 4, 5, 6, 7], and _step_ are vectors, then you could rewrite as _base_ + _i_ * _step_.

Reply:
Good point.